### ISSUE TITLE
Infinite digest loop error logs too much data

### ISSUE THREAD

**Issue by adamreisnz** [angular/angular.js#15043]
**Do you want to request a _feature_ or report a _bug_?**
Enhancement

**What is the current behavior?**
![image](https://cloud.githubusercontent.com/assets/490562/17795682/7793f2c8-660e-11e6-8d70-69d2bbf6cb9c.png)

This goes on for pages and pages.

**What is the expected behavior?**
I would like the error message to be more legible, and for it not to include all my products JSON including Lorem Ipsum text etc. in it, because it's not being helpful and I can't make out anything meaningful from that mess. 

**What is the motivation / use case for changing the behavior?**
To be able to debug more efficiently.

**Which versions of Angular, and which browser / OS are affected by this issue? Did this work in previous versions of Angular? Please also test with the latest stable and snapshot (https://code.angularjs.org/snapshot/) versions.**
1.5.8

**Other information (e.g. stacktraces, related issues, suggestions how to fix)**
Don't output nested object information, I don't think this is necessary. Or at least give us a flag so we might be able to toggle it on or off? Open to other suggestions as well.

**Comment by gkalpak** [comment]
First of all, these error messages are not supposed to be legible in the console - they are URL encoded so the data can be presented better in the error page that the URL points to.

But I suppose logging arbitrarily deep objects doesn't help either. Maybe we could have a (configurable?) limit in the depth in which we traverse the logged objects.

**Comment by adambuczynski** [comment]
@gkalpak Yeah, I think I gave up on clicking those links a long time ago as they often didn't work or  didn't properly show the full extent of the error. There are some open issues in the Angular repo around this from what I saw yesterday when I was searching for my issue, so I don't find it very reliable.

Second, I prefer debugging in the Chrome console, and I often need to inspect network requests and the like, so I have it open most of the time. These monstrously long error messages are occasionally freezing up my console or almost slowing it down to a halt. And I have a pretty high end computer ;)

I just don't think that such long messages are helping anyone. The error seems to be chucking whatever object/array that changed into the log, which could very well be an array of 1000 complex objects that was received from the server.

I think some depth limit should definitely be applied, and for arrays perhaps a limit of how many items are displayed out of that array with something like `...295 more` at the end.
